type-annotator-scanner/src/main/java/edu/ucr/cs/riple/scanner/TypeAnnotatorScanner.java

```diff
@@ -106,9 +106,10 @@ public Description matchMethod(MethodTree tree, VisitorState state) {
       return Description.NO_MATCH;
     }
     Symbol.MethodSymbol methodSymbol = ASTHelpers.getSymbol(tree);
-    MethodInfo methodInfo = MethodInfo.findOrCreate(methodSymbol, state, context);
+    MethodInfo methodInfo = MethodInfo.findOrCreate(methodSymbol, context);
     methodInfo.findParent(state, context);
     methodInfo.setAnnotation(config);
+    methodInfo.setURI(state);
```

Comment:
So, I understand why we are adding the check for a `null` `tree.getSourceFile()` (generated classes and so on, I assume), but why are we moving this computation from the constructor to this `setURI(...)` method? Do we expect his to be called multiple times? Or a long time after first calling `MethodInfo.findOrCreate(...)`?

It seems more natural to me that `MethodInfo.uri` is `final` and set by the constructor.

Either way, that's partly a matter of style, so I am not opposed to either way.

Reply:
Yes, the problem is that the actual call to `setURI` is called after the initial object creation via `findOrCreate`. That happens when a sub-method is visited sooner than the the super-method and this is the reason we were having wrong `URI`s in our nodes.

Reply:
Please note, that while visiting a `sub-method` a super method node is `created / reserved` and then we fill other info for that node when we are visiting it via `match` methods.

Reply:
@lazaroclapp I checked adding a unit test with mocking for this case. It is not straightforward adding this test without changing the constructors (Requires changing the return result of `CompilationUnitTree` passed from Error Prone). Considering that the objective of this PR is not handling `tree.getSourceFile()` returning `null` but passing the correct `VisitorState` to method node (not the `VIsitorState` pointing to super-method). Would you please tell me if I should still try to make a unit test for this ? Thank you.

Reply:
Ah, I think I understand this better now. `findOrCreate(...)` is called both here and also from `findParent(...)`, but when called from `findParent(...)` the `VisitorState` will correspond to the overridding method, while the `methodSymbol` will be the method being overridden. Moving the logic of `setURI(...)` to a separate call that only happens when directly visiting the given method (via `matchMethod`), ensures that the `VisitorState` is looking at the file where `methodSymbol` is declared. Correct?

Reply:
@lazaroclapp Yes exactly. This is the issue that has been resolved.